app/controllers/listings_controller.rb

```diff
@@ -0,0 +1,10 @@
+class ListingsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
Both your listings and my_listings controllers are authenticating that the user is logged in. I also imagine many of your other pages will have the check that you're logged in as well. For this reason we tend to put the `authenticate_user!` callback in the `ApplicationController`. This will make it so all your controllers by default will require you to be logged in. Then if for some reason you don't want a controller to require someone to be logged in you can use `skip_callback` on the specific controller.